common/lib/capa/capa/capa_problem.py

```diff
@@ -808,7 +825,13 @@ def _extract_html(self, problemtree):  # private
         # otherwise, render children recursively, and copy over attributes
         tree = etree.Element(problemtree.tag)
         for item in problemtree:
-            item_xhtml = self._extract_html(item)
+            item_xhtml = self._extract_html(item, question_id)
+
+            if item.tag == 'question':
+                item_xhtml.set('class', 'question')
+                item_xhtml.set('id', str(question_id))
```

Comment:
Great!   

RE: inheritance 

I'm hoping we can expose  the value of ```question_id``` to the child templates because they will need to be able to add their own unique ids like this:

```
<div class="problem">
    <div>
        <div class="question", id="question-0">

</div>
```